plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -0,0 +1,210 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Admin\Features\Features;
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * CatalogSorting class.
+ */
+class AddToCartWithOptions extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'add-to-cart-with-options';
+
+	/**
+	 * Initializes the AddToCartWithOptions block and hooks into the `wc_add_to_cart_message_html` filter
+	 * to prevent displaying the Cart Notice when the block is inside the Single Product block
+	 * and the Add to Cart button is clicked.
+	 *
+	 * It also hooks into the `woocommerce_add_to_cart_redirect` filter to prevent redirecting
+	 * to another page when the block is inside the Single Product block and the Add to Cart button
+	 * is clicked.
+	 *
+	 * @return void
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_filter( 'wc_add_to_cart_message_html', array( $this, 'add_to_cart_message_html_filter' ), 10, 2 );
+		add_filter( 'woocommerce_add_to_cart_redirect', array( $this, 'add_to_cart_redirect_filter' ), 10, 1 );
```

Comment:
Good question. Probably at the Button level it would be better, no? But I'm fine leaving it here for now and moving it to the Button block in the follow-up PR.

Reply:
Let's tackle this as part of issue https://github.com/woocommerce/woocommerce/issues/53010